crates/epaint/src/text/text_layout.rs

```diff
@@ -198,28 +201,17 @@ fn line_break(
     let mut non_empty_rows = 0;
 
     for i in 0..paragraph.glyphs.len() {
-        let potential_row_width = paragraph.glyphs[i].max_x() - row_start_x;
+        let potential_row_width = paragraph.glyphs[i].max_x() - row_start_x - first_row_indentation;
 
         if job.wrap.max_rows > 0 && non_empty_rows >= job.wrap.max_rows {
             break;
         }
 
-        if potential_row_width > job.wrap.max_width {
-            if first_row_indentation > 0.0
-                && !row_break_candidates.has_good_candidate(job.wrap.break_anywhere)
-            {
-                // Allow the first row to be completely empty, because we know there will be more space on the next row:
```

Comment:
If `first_row_indentation == 0.0` then we never want an empty first row. We would rather overflow `max_width`, because otherwise we will just run into the same problem on the next row. For instance if `max_width == 0.0` we want a layout where we have exactly one glyph per row.

If `first_row_indentation > 0.0` however we are fine with having an empty row, because there will be more space available on the next row, and that might be enough to fit the first word or whatever.

Reply:
I just ran this in examples/wrapped_layout and it seems to work as you describe (one glyph per row).

```
let mut chui = ui.child_ui(
    Rect::from_two_pos(ui.cursor().min, ui.cursor().min),
    Layout::left_to_right(Align::Center),
);
chui.horizontal_wrapped(|ui| {
    ui.label("Frick");
});
```